src/Altinn.Apps/AppTemplates/AspNet/App.IntegrationTests/Mocks/Services/DataMockSI.cs

```diff
@@ -148,7 +149,34 @@ public Task<DataElement> UpdateBinaryData(string org, string app, int instanceOw
 
         public Task<DataElement> UpdateData<T>(T dataToSerialize, Guid instanceGuid, Type type, string org, string app, int instanceOwnerId, Guid dataId)
         {
-            throw new NotImplementedException();
+            string dataPath = GetDataPath(org, app, instanceOwnerId, instanceGuid);
+
+            Instance instance = GetTestInstance(app, org, instanceOwnerId, instanceGuid);
+
+            DataElement dataElement = instance.Data.FirstOrDefault(r => r.Id.Equals(dataId.ToString()));
+
+            try
+            {
```

Comment:
Probably best to let the logic fail with an exception. Easier to analyse the reason with a stack trace.

Reply:
fixed